### DISCUSSION TITLE
Tutorial to get daily dashboard summary by email

### DISCUSSION THREAD

**Discussion by b-u-g-s** [Security-Onion-Solutions/securityonion#8499] [General]
Hi everyone,

This is a long shot, as my searches did not returned anything. 
I was wondering if any kind souls around here could please point me to a tutorial I could follow to get email alerts.

Below I will describe:
1. What I would like to achieve. 
2. Provide a bit more context as to why, this could be useful (and not just for me)
3. The search I did to find some information and where I got stuck4. 
4. What I did in the past (back in the day of SO 1.xxx ?)

---
1. What I would like to achieve. 
To get a daily e-mail with a summary of all the alerts that have not been acknowledged on the ALERT page. Basically, to have all the information that the ALERT page have (with a 24h window)

2. Provide a bit more context as to why, this could be useful (and not just for me)
In a scenario of a small business running SO, there might be only a few people looking at the SO dashboard and maybe not every day. So it would useful to get a daily summary of what new alerts have come up and if you should log on to SO for further investigation.
In my case, I have a small office and there is only 3x people working so not too much network traffic. but there is a lot of Internet traffic from IOT, labs, etc. I usually look at SO once a week, and clear all the alerts. When I see something I don't expect, I investigate further. That's for our internal day to day security.

I also sometimes do incident investigation, where I might have a compromised client's laptop and depending of the threat I am assessing, I might let the laptop running in a DMZ where I have a SO analysing the traffic coming out. In this case, I may look at it every hour or so to see if there is anything weird. But I had a few cases where I had to let it run for a week to be sure there wasn't any "call back, etc". This is done in parallel of doing a forensic investigation of an offline copy of the hard drive, most of the time this is where you find the real stuff, but I found having SO running in the background when you are allowed/can let the asset you are investigating turned on can give you some additional info and even find stuff you may have missed.

Lastly, at some client's places where they use SO, I like to check up now and then if the teams are clearing their alerts, etc.
If I could just get an email (once a day at the beginning, once a week later, etc) that gives me a summary of what is left unattended, could give me a sense on how they are doing. I understand it doesn't tell me if they have investigated the alert correctly, but that's a start.

3. The search I did to find some information and where I got stuck 
I saw the documentation for elastalert here:
https://docs.securityonion.net/en/2.3/elastalert.html
and then here:
https://elastalert2.readthedocs.io/en/latest/elastalert.html#overview

It looks to me as being maybe quite generic.
I tried creating a rule for Cardinality (I am not even sure what each of those alert types are for, even if I can guess but it doesn't seem to be explained in the SO doc). And I didn't really know which of the 2x index patterns to use... so I selected the so-ids.
But then it was asking me for what fields I wanted to use... there was no list of what I could enter (maybe because there are too many).
In the end I entered what I could, just pressed enter (empty values) for some others (no error), but then I did a so-elastalert-test and that failed (in the file created I got a "traceback (most recent call last)... error"
Probably because I entered some empty values

The point is... I think what I am trying to do is not difficult, just a daily email with all the unacknowledged alerts
But I was wondering if anyone had already done that before? 
Like an idiot guide to setup a daily email with the top issues detected in the last 24 hours.

Because if there was, that would be great, I wouldn't have to RTFM :)

4. What I did in the past (back in the day of SO 1.xxx ?)
Quite a long time ago, probably 6 or 7 years ago, with one of the first version of SO, I don't think there was elastalert.
I did something like that, but it was more of a "dirty hack".
Basically, I setup the dashboard I liked with:
- Top 10 alerts from the last 24h
- Source IP and Destination IP (very important so I could quickly see if it was going/coming from an important IP or subnet)

and every day at 7pm, I had script running through cron on the SO that was connecting to the SO web portal, taking a screenshot and sending it by email. I can't exactly remember what set of tools I was using to do this, I think one of them was something like wkhtml or something like that to create a PDF.
Then I stopped using SO for a few years, and when I started again I never looked at using this again.

I don't think it was a "pretty" solution, so when I saw the elastalert section of SO I got hopeful, but then got a bit scared at the fact it doesn't seem very straightforward (to me at least).

I remember also it was super useful once because there was a specific IDS alert that was coming from a phone I was using and I couldn't find out what was triggering that alert, but every single day I could see it in my alert dashboard, coming from the same IP (my phone). I changed my script to only send me an email if that alert was detected, and set a 1 minutes email check. This allowed me to pin point which app that was attempting to contact that bad IP. It was an app I was using everyday but at different time, but as there was a few seconds delay for the alert to appear in the SO dashboard and it appeared random (at the time I think it was a minute or so actually, that's why it was difficult to identify)

Anyway, would be nice to have an easy email alert system or a simple tutorial to follow :)
Thanks for reading
Bugs.

**Comment by b-u-g-s** [comment]
Forgot to mention that I used so-elastalert-create for creating the rules

**Comment by defensivedepth** [comment]
The community member in this discussion might have some ideas: https://github.com/Security-Onion-Solutions/securityonion/discussions/8393

**Reply by b-u-g-s** [reply]
Thank you for the link, will look into that!
